pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -128,114 +155,149 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{}, nil
 		}
 
-		log.Error(err, errGetES)
+		log.Error(err, logErrorGetES)
 		syncCallsError.With(resourceLabels).Inc()
-
-		return ctrl.Result{}, err
-	}
-
-	// See https://github.com/external-secrets/external-secrets/issues/3604
-	// We fetch the ExternalSecret resource above, however the status subresource is inconsistent.
-	// We have to explicitly fetch it, otherwise it may be missing and will cause
-	// unexpected side effects.
-	err = r.SubResource("status").Get(ctx, &externalSecret, &externalSecret)
-	if err != nil {
-		log.Error(err, "failed to get status subresource")
 		return ctrl.Result{}, err
 	}
 
-	timeSinceLastRefresh := 0 * time.Second
-	if !externalSecret.Status.RefreshTime.IsZero() {
-		timeSinceLastRefresh = time.Since(externalSecret.Status.RefreshTime.Time)
-	}
-
 	// skip reconciliation if deletion timestamp is set on external secret
-	if externalSecret.DeletionTimestamp != nil {
-		log.Info("skipping as it is in deletion")
+	if !externalSecret.GetDeletionTimestamp().IsZero() {
+		log.V(1).Info("external secret is marked for deletion, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
 	// if extended metrics is enabled, refine the time series vector
 	resourceLabels = ctrlmetrics.RefineLabels(resourceLabels, externalSecret.Labels)
 
 	if shouldSkipClusterSecretStore(r, externalSecret) {
-		log.Info("skipping cluster secret store as it is disabled")
+		log.V(1).Info("skipping cluster secret store as it is disabled")
 		return ctrl.Result{}, nil
 	}
 
 	// skip when pointing to an unmanaged store
 	skip, err := shouldSkipUnmanagedStore(ctx, req.Namespace, r, externalSecret)
```

Comment:
The error here is not handled.

Reply:
I have fixed this in https://github.com/external-secrets/external-secrets/pull/4086/commits/a96f389976c4b1c0a3fc081914cf621fdc97cd75

But it caused some test failures, because it was stopping the reconcile at that point when the generators/secretstores did not exist.

And that was causing the tests which expect the ES to go into an error condition not to succeed (because we dont update the ES status at this stage).

To fix that, I just made that `shouldSkipUnmanagedStore` function ignore SecretStores/Generators which dont exist.

Reply:
Heh, nice. :)